AndroidCompat/Config/src/main/java/xyz/nulldev/ts/config/Logging.kt

```diff
@@ -66,16 +71,44 @@ private fun getLogger(name: String): ch.qos.logback.classic.Logger {
     return context.getLogger(name)
 }
 
-fun initLoggerConfig(appRootPath: String) {
+fun initLoggerConfig(
+    appRootPath: String,
+    maxFiles: Int,
+    maxFileSize: String,
+    maxTotalSize: String,
+) {
     val context = LoggerFactory.getILoggerFactory() as LoggerContext
     val logger = getBaseLogger()
+
     // logback logs to the console by default (at least when adding a console appender logs in the console are duplicated)
-    logger.addAppender(createRollingFileAppender(context, "$appRootPath/logs"))
+    logger.addAppender(createRollingFileAppender(context, "$appRootPath/logs", maxFiles, maxFileSize, maxTotalSize))
 
     // set "kotlin exposed" log level
     setLogLevelFor("Exposed", Level.ERROR)
 }
 
+fun updateFileAppender(
+    maxFiles: Int,
+    maxFileSize: String,
+    maxTotalSize: String,
+) {
+    val logger = getBaseLogger()
+
+    val appender = logger.getAppender(FILE_APPENDER_NAME) as RollingFileAppender<*>? ?: return
+    val rollingPolicy = appender.rollingPolicy as SizeAndTimeBasedRollingPolicy<*>
+    rollingPolicy.apply {
+        maxHistory = maxFiles
+        setMaxFileSize(FileSize.valueOf(maxFileSize))
```

Comment:
How forgiving is this? I don't want to cause a server crash if a client sends a invalid value like `10gbb`

Reply:
it did not crash when sending the mutation, however, once an invalid value was in the config file the server crashed on startup

Reply:
Maybe make a extension function that wraps FileSize.valueOf in a try/catch with a default if its invalid, I want to protect against manual file editing errors as well

Reply:
because it's a java class a companion extension function can't be added